Library/DiscUtils.Ext/ExtentsFileBuffer.cs

```diff
@@ -70,36 +70,46 @@ public override int Read(long pos, byte[] buffer, int offset, int count)
 
             while (totalBytesRemaining > 0)
             {
-                uint logicalBlock = (uint)((pos + totalRead) / blockSize);
-                int blockOffset = (int)(pos + totalRead - logicalBlock * blockSize);
+                uint logicalBlock = (uint)((pos + totalRead) / blockSize);  // logical block containing the next byte to read
+                int blockOffset = (int)(pos + totalRead - logicalBlock * blockSize);  // offset within 'logicalBlock' of the next byte to read
 
                 int numRead = 0;
 
+                // Find the extent containing 'logicalBlock' or, if none, the first extent beyond it.
                 Extent extent = FindExtent(extents, logicalBlock);
+
                 if (extent == null)
                 {
-                    throw new IOException("Unable to find extent for block " + logicalBlock);
+                    // The remainder of the sparse file is one big hole; all bytes remaining to be read in the file are zeroes.
```

Comment:
Did you hit this after fixing the high bit issue in file Extent.cs?  This looks like a real bug.